### PR TITLE
feature(util): new utility to create http params

### PR DESCRIPTION
**Pull request by nappynapster**
@plentymarkets/team-terra

#### Must be done by Terra

Testing
- [ ] Person 1 (mandatory)
- [ ] Person 2 (mandatory)

Information transfer
- [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Browser-Support (relevant for changes in scss / appearance of components and jQuery)
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Terra Basic Plugin
- [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

----

#### Must be done by every developer
Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline). 

Documentation
- [x] Changelog (optional: relevant for every change which influences the API)
- [ ] Example (updated or created)
- [ ] JSDoc (optional: relevant for every change which influences the API)

Testing
- [ ] Unit Test (updated or created)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <a name="5.0.0"></a>
 # 5.0.0 (Unreleased)
 
+## Features
+* **createHttpParams** new createHttpParams function is the new replacement for the deprecated createUrlSearchParams from TerraBaseService function.
```
Comment (ChristianMoll) [inline CHANGELOG.md]: ein new kannst du dir denke ich sparen :D

Comment (nappynapster) [reply CHANGELOG.md]: :D ja

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <a name="5.0.0"></a>
 # 5.0.0 (Unreleased)
 
+## Features
+* **createHttpParams** new createHttpParams function is the replacement for the deprecated createUrlSearchParams from TerraBaseService.
```
Comment (pweyrich) [inline CHANGELOG.md]:
```suggestion
* **utils** provide new `createHttpParams()` method which replaces the deprecated `createUrlSearchParams()` of the `TerraBaseService`.
```

File: src/lib/utils/terra-http-params.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    HttpParams,
+    HttpUrlEncodingCodec
+} from '@angular/common/http';
+import {
+    isArray,
+    isNullOrUndefined
+} from "util";
+import { TerraKeyValueInterface } from '../models';
+
+export function createHttpParams(params:TerraKeyValueInterface<any>, arrayAsArray:boolean = false):HttpParams
```
Comment (pweyrich) [inline src/lib/utils/terra-http-params.ts]:
```suggestion
export function createHttpParams(params:Params, arrayAsArray:boolean = false):HttpParams
```

File: src/lib/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { createHttpParams } from './terra-http-params';
```
Comment (pweyrich) [inline src/lib/utils/index.ts]: Das muss auch noch in der public-api exportiert werden, sonst kann es nicht verwendet werden

**Comment by pweyrich** [comment]
Replaced by https://github.com/plentymarkets/terra-components/pull/1047

**Pull request closed without merging**